### PR TITLE
Favorite a post on town square

### PR DESCRIPTION
**Pull request by zoek1**
##### Description

**Expected Behavior**
- [X] click a star to toggle favorite post.
- [X] Adds to my favorites on filters (left side)

**Definition of Done**
- [X] Extend and use this favorite api PR #4751
- [X] add ability to favorite on townsquare (see location at bottom screenshot)
- [X] add ability to filter and see all the favorited post
- [X] add ability to unfavorite posts
- [X] post a video for core team review

##### Refers/Fixes

#6121

##### Testing
https://www.dropbox.com/s/rc7g19lk1t0wptw/2020-03-09%2005-09-33.flv?dl=0

### CODE REVIEW

**Comment by zoek1** [comment]
Ready for review :smiley:

**Comment by octavioamu** [comment]
Hey Zoek how is this using #4751 ?

**Comment by zoek1** [comment]
Hey @octavioamu, i looked at #4751 but it has some unnecessary things and others are already superseded by the Townsquare development.

Unnecessary:
 - All code related to adding one app only for favorite functionality

Superseded:
- Actually the activity model has links to bounties, grants, users, kudos, etc. So linking the activity object and one user performs the same task and is more general.

Made on both PR:
- Create a favorite model
- Add option to add or delete favorite
- List of favorites

**Comment by zoek1** [comment]
@danlipert @octavioamu @thelostone-mc is it possible to integrate this into the next release? @PixelantDesign

File: app/townsquare/views.py [hunk 1]
```diff
@@ -466,6 +481,15 @@ def api(request, activity_id):
         if request.POST['direction'] == 'unliked':
             activity.likes.filter(profile=request.user.profile).delete()
 
+    # like request
+    elif request.POST.get('method') == 'favorite':
+        if request.POST['direction'] == 'favorite':
+            already_likes = Favorite.objects.filter(activity=activity, user=request.user).exists()
+            if not already_likes:
+                Favorite.objects.create(user=request.user, activity=activity)
+        if request.POST['direction'] == 'unfavorite':
```
Comment (danlipert) [inline app/townsquare/views.py]: should be else if here

Comment (zoek1) [reply app/townsquare/views.py]: done! :)

**Comment by owocki** [comment]
playing with this in prod right now. looking good!

**Pull request merged**